enterprise/reporting/reporting.go

```diff
@@ -678,11 +675,11 @@ func (r *DefaultReporter) Report(ctx context.Context, metrics []*types.PUReporte
 		return nil
 	}
 
-	maxLength := r.eventNameMaxLength.Load()
 	prefixLength := r.eventNamePrefixLength.Load()
 	suffixLength := r.eventNameSuffixLength.Load()
-	if prefixLength <= 0 || suffixLength <= 0 || prefixLength >= maxLength || suffixLength >= maxLength || prefixLength+suffixLength != maxLength {
-		err := fmt.Errorf("invalid event name trimming configuration: prefixLength=%d, suffixLength=%d, maxLength=%d. prefixLength and suffixLength must be > 0, prefixLength < maxLength, suffixLength < maxLength, and prefixLength + suffixLength = maxLength", prefixLength, suffixLength, maxLength)
+	maxLength := prefixLength + suffixLength
+	if prefixLength <= 0 || suffixLength <= 0 {
+		err := fmt.Errorf("invalid event name trimming configuration: prefixLength=%d, suffixLength=%d. prefixLength and suffixLength must be > 0", prefixLength, suffixLength)
```

Comment:
Can't we just return a generic error and log `prefixLength` and `suffixLength`?
If yes we can avoid using reflection (i.e. `fmt.Errorf`) and go with the non-sugared logger flow:

```go
r.log.Errorn(`[ Reporting ]: Invalid event name trimming configuration`, 
    obskit.Error(err),
    logger.NewIntField("prefixLength", prefixLength),
    logger.NewIntField("suffiexLength", suffixLength),
)
```